astropy/utils/data.py

```diff
@@ -1001,9 +1001,11 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
     # Check if URL is Astropy data server, which has alias, and cache it.
     if (url_key.startswith(conf.dataurl) and
             conf.dataurl not in _dataurls_to_alias):
-        with urllib.request.urlopen(conf.dataurl, timeout=timeout) as remote:
-            _dataurls_to_alias[conf.dataurl] = [conf.dataurl, remote.geturl()]
-
+        try:
+            with urllib.request.urlopen(conf.dataurl, timeout=timeout) as remote:
+                _dataurls_to_alias[conf.dataurl] = [conf.dataurl, remote.geturl()]
+        except urllib.error.URLError:  # Host unreachable
+            _dataurls_to_alias[conf.dataurl] = [conf.dataurl]
```

Comment:
```suggestion
           msg = ('url "{}" was requested in download_file, but was not found.'
                        ' Will try cached file if it exists.')
           warn(AstropyWarning(msg.format(conf.dataurl)))
            _dataurls_to_alias[conf.dataurl] = [conf.dataurl]
```

Reply:
@eteq , personally, I would prefer *not* emitting a warning. This is because this line was only to cache an alias of the original URL. Hence, user will not find the warning make any sense unless user knows the internal mechanics of this function. If this warning turns out to be necessary, we can always follow up in a future PR. Thank you for your understanding!